File: app/src/main/java/ca/gainzassist/models/db/SessionDao.java
package ca.gainzassist.models.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import androidx.lifecycle.LiveData;
import ca.gainzassist.models.Session;
import java.util.List;

@Dao
public interface SessionDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Long insert(Session session);


    @Query("SELECT * from  sessions")
    LiveData<List<Session>> getAll();
//    @Query("SELECT id FROM  sessions WHERE timestamp = :timestamp")
//    int getId(long timestamp);

//    @Query("SELECT * FROM sets WHERE exercise_id = :exerciseId")
//    LiveData<List<ExerciseSet>> getLiveFromExercise(int exerciseId);
//
//    @Query("SELECT * FROM sets WHERE exercise_id = :exerciseId")
//    List<ExerciseSet> getFromExercise(int exerciseId);


    @Update
    void update(Session... sessions);


    @Delete
    void delete(Session... sessions);
}

File: app/src/main/java/ca/gainzassist/data/local/database/SessionDao.kt
package ca.gainzassist.data.local.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import ca.gainzassist.domain.model.Session

@Dao
interface SessionDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(session: Session): Long

    @Query("SELECT * from  sessions")
    fun getAll(): LiveData<List<Session>>

    @Update
    fun update(vararg sessions: Session)

    @Delete
    fun delete(vararg sessions: Session)
}
